src/dynamics/fixed_points/basic.lean

```diff
@@ -71,6 +71,24 @@ calc fb (g x) = g (fa x) : (h.eq x).symm
 protected lemma apply {x : α} (hx : is_fixed_pt f x) : is_fixed_pt f (f x) :=
 by convert hx
 
+@[simp] lemma _root_.set.preimage_iterate_eq {f : α → α} {n : ℕ} :
+  set.preimage (f^[n]) = ((set.preimage f)^[n]) :=
```

Comment:
```suggestion
  set.preimage (f^[n]) = (set.preimage f)^[n] :=
```
or does this fail because of weird precedence?

Reply:
Unfortunately these annoying brackets are necessary because of "weird precedence".